native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/WriteArgsFileMojo.java

```diff
@@ -67,6 +71,24 @@ public class WriteArgsFileMojo extends NativeCompileNoForkMojo {
     @Override
     public void execute() throws MojoExecutionException {
         List<String> args = getBuildArgs();
+
+        getLog().info("Cleaning old native image build args");
+
+        try (Stream<Path> listStream = Files.list(outputDirectory.toPath())) {
+            listStream.map(path -> path.getFileName().toString())
+                    .filter(f -> f.startsWith("native-image") && f.endsWith("args"))
+                    .map(outputDirectory.toPath()::resolve)
+                    .forEach(file -> {
+                        try {
+                            Files.delete(file);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
```

Comment:
Also, you should probably throw `MojoExecutionException` here

Reply:
in forEach it only allows `RuntimeException` to be thrown